packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -132,3 +110,41 @@ export const SelectedItemCheck = styled( Ariakit.SelectItemCheck )`
 	margin-inline-start: ${ ITEM_PADDING };
 	font-size: 24px; // Size of checkmark icon
 `;
+
+const truncateStyles = css`
+	overflow: hidden;
+	text-overflow: ellipsis;
+	white-space: nowrap;
+`;
+
+export const ItemHintWrapper = styled.div`
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	flex-wrap: wrap;
+	flex: 1;
+`;
+
+export const SelectedExperimentalHintWrapper = styled.div`
+	${ truncateStyles }
+`;
+
+export const SelectedExperimentalHintItem = styled.span`
+	color: ${ COLORS.theme.gray[ 600 ] };
+	margin-inline-start: ${ space( 2 ) };
+`;
+
+export const ItemHintContent = styled.span`
+	padding-inline-end: ${ space( 4 ) };
+`;
+
+export const ItemHint = styled.span`
+	// Necessary to override V1 styles passed via the legacy
+	// '.components-custom-select-control__item-hint' class
+	${ SelectItem } && {
+		color: ${ COLORS.theme.gray[ 600 ] };
+		text-align: initial;
```

Comment:
Since we use flexbox to push the hint to the right, `text-align` isn't necessary _per se_, but since V1 styles are "leaking" to this implementation too, we need to reset the value to `initial`. This allows the hint text to be correctly aligned both when it's on the same line as the item's content, and when it wraps to its own independent line.